src/wp-includes/class-wp-hook.php

```diff
@@ -299,6 +309,48 @@ public function apply_filters( $value, $args ) {
 		return $value;
 	}
 
+	/**
+	 * Gets the the invalid callback error message.
+	 *
+	 * @since 5.6.0
+	 *
+	 * @param mixed $callback Invalid callback.
+	 *
+	 * @return string error message.
+	 */
+	protected function get_invalid_callback_error_message( $callback ) {
+		return sprintf(
+			/* translators: %s: Invalid callback name or type. */
+			__( 'Requires <code>%s</code> to be a valid callback.' ),
+			$this->convert_callback_to_string( $callback )
+		);
+	}
+
+	/**
+	 * Converts the given callback into a string.
+	 *
+	 * @since 5.6.0
+	 *
+	 * @param mixed $callback Callback to convert into string.
+	 *
+	 * @return string callback as a string.
+	 */
+	protected function convert_callback_to_string( $callback ) {
+		if ( is_string( $callback ) ) {
+			return $callback;
+		}
+
+		if ( ! is_array( $callback ) ) {
+			return gettype( $callback );
+		}
+
+		if ( is_object( $callback[0] ) ) {
```

Comment:
I think we might need to stringify an empty arrays and associative arrays (`$callback = ['name' => 'test'])` and `$callback = array()`